engine/src/main/java/com/datatorrent/stram/plan/physical/PhysicalPlan.java

```diff
@@ -979,6 +998,14 @@ public void deployChanges() {
     this.undeployOpers.removeAll(newOpers.keySet());
```

Comment:
If I understand @tweise correctly, he is suggesting that the existing code be augmented _only_ by the highlighted fragment below (since he says "remove everything added new"):

{code}
    updatePartitionsInfoForPersistOperator(this.dag);

    /* begin new code */
    // redeploy dependencies of the new operators excluding the new operators themselves
    this.undeployOpers.addAll(getDependents(newOpers.keySet()));
    /* end new code */

    this.undeployOpers.removeAll(newOpers.keySet());
    //make sure all the new operators are included in deploy operator list
    this.deployOpers.addAll(this.newOpers.keySet());
    // include downstream dependencies of affected operators into redeploy
    Set<PTOperator> deployOperators = this.getDependents(this.deployOpers);
    ctx.deploy(releaseContainers, this.undeployOpers, newContainers, deployOperators);
{code}

This works for _undeployOpers_ since the argument set (newOpers) is removed just after the new code. However, this change alone does not add dependents to _deployOpers_ as the proposed fix does -- not sure if this makes a difference.